lib/elixir/lib/uri.ex

```diff
@@ -221,12 +225,12 @@ defmodule URI do
       true
 
   """
-  @spec char_unreserved?(term) :: boolean
-  def char_unreserved?(c) do
-    c in ?0..?9 or
-    c in ?a..?z or
-    c in ?A..?Z or
-    c in '~_-.'
+  @spec char_unreserved?(pos_integer) :: boolean
+  def char_unreserved?(char) when is_integer(char) and char > 0 do
+    char in ?0..?9 or
+    char in ?a..?z or
+    char in ?A..?Z or
+    char in '~_-.'
   end
```

Comment:
Seem clearer to me.

Do both perform the same, or is the guard one faster due to some VM guard optimisation?